src/liballoc/boxed.rs

```diff
@@ -63,6 +63,27 @@
 //! T` obtained from `Box::<T>::into_raw` may be deallocated using the
 //! [`Global`] allocator with `Layout::for_value(&*value)`.
 //!
+//! `Box<T>` has the same ABI as `&mut T`. In particular, when `T: Sized`,
+//! this allows using `Box<T>` in FFI:
+//!
+//! ```c
+//! /* C header */
```

Comment:
IMO, `foo`/`bar` examples are not so nice. How about something a bit more semantic, like

```c
struct Foo;
struct Foo* foo_new(void);
void foo_destroy(struct Foo*);
```

Reply:
Thank you for the review! I updated the example.